### PR TITLE
Add `swift-bootstrap` command

### PR DESCRIPTION
**Pull request by neonichu**
This command can be used to bootstrap SwiftPM via CMake without requiring us to build a full-blown SwiftPM using CMake.

### CODE REVIEW

**Comment by neonichu** [comment]
@swift-ci please smoke test

File: Utilities/bootstrap [hunk 1]
```diff
@@ -584,6 +584,7 @@ def build_swiftpm_with_cmake(args):
         "-DSwiftCrypto_DIR="      + os.path.join(args.build_dirs["swift-crypto"],          "cmake/modules"),
         "-DSwiftSystem_DIR="      + os.path.join(args.build_dirs["swift-system"],          "cmake/modules"),
         "-DSwiftCollections_DIR=" + os.path.join(args.build_dirs["swift-collections"],     "cmake/modules"),
+        "--target", "swift-bootstrap",
```
Comment (neonichu) [inline Utilities/bootstrap:587]: This doesn't seem to actually work, I can see it in the invocation, but other targets are still being built, e.g. the other executables.

Comment (neonichu) [reply Utilities/bootstrap:587]: Ah, since we don't actually build using CMake, we need to do the target filtering via `ninja`.

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
Failures look as if we're relying on the first-stage to produce the `PackagePlugin` module and we don't use the one provided by the second stage?

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Review by tomerd — APPROVED** [review]

**Comment by neonichu** [comment]
@swift-ci please smoke test

**Comment by neonichu** [comment]
@swift-ci please smoke test macOS

**Pull request merged**